File: src/types/content.ts
import type { OutputFormat, TonePreset, LengthPreset, PlatformConfig } from './platform';

export interface ContentInput {
  id: string;
  title: string | null;
  source_url: string | null;
  raw_text: string;
  word_count: number;
  created_at: string;
}

export interface FetchedContent {
  title: string | null;
  text: string;
  word_count: number;
}

export interface KeyPoints {
  main_thesis: string;
  key_arguments: string[];
  supporting_data: string[];
  target_audience: string;
  emotional_tone: string;
  call_to_action: string | null;
}

export interface RepurposedOutput {
  id: string;
  content_input_id: string;
  format: string;
  output_text: string;
  created_at: string;
}

export interface RepurposeRequest {
  content: string;
  source_url?: string;
  title?: string;
  formats: OutputFormat[];
  tone: TonePreset;
  length: LengthPreset;
  voice_id?: string;
  config?: PlatformConfig;
}

export interface RepurposeResponse {
  content_input_id: string;
  outputs: RepurposedOutput[];
}

export interface TwitterThreadData {
  tweets: string[];
}

export interface EmailSequenceData {
  emails: Array<{
    email_number: number;
    label: string;
    subject_line: string;
    preview_text: string;
    body: string;
    cta_text: string;
  }>;
}

export interface NewsletterData {
  subject_line: string;
  preview_text: string;
  body: string;
}

export interface HistoryItem {
  id: string;
  title: string | null;
  word_count: number;
  format_count: number;
  created_at: string;
}

export interface HistoryPage {
  items: HistoryItem[];
  total: number;
  page: number;
  page_size: number;
}

export interface HistoryDetail {
  input: ContentInput;
  outputs: RepurposedOutput[];
}

File: src-tauri/src/models/content.rs
use serde::{Deserialize, Serialize};

use super::platform::OutputFormat;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ContentInput {
    pub id: String,
    pub source_url: Option<String>,
    pub raw_text: String,
    pub title: Option<String>,
    pub word_count: u32,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct KeyPoints {
    pub main_thesis: String,
    pub key_arguments: Vec<String>,
    pub supporting_data: Vec<String>,
    pub target_audience: String,
    pub emotional_tone: String,
    pub call_to_action: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RepurposedOutput {
    pub id: String,
    pub content_input_id: String,
    pub format: String,
    pub output_text: String,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HistoryPage {
    pub items: Vec<HistoryItem>,
    pub total: u32,
    pub page: u32,
    pub page_size: u32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HistoryItem {
    pub id: String,
    pub title: Option<String>,
    pub word_count: u32,
    pub format_count: u32,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HistoryDetail {
    pub input: ContentInput,
    pub outputs: Vec<RepurposedOutput>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FetchedContent {
    pub title: Option<String>,
    pub text: String,
    pub word_count: u32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RepurposeRequest {
    pub content: String,
    pub source_url: Option<String>,
    pub title: Option<String>,
    pub formats: Vec<OutputFormat>,
    pub tone: super::platform::TonePreset,
    pub length: super::platform::LengthPreset,
    pub voice_id: Option<String>,
    pub config: Option<super::platform::PlatformConfig>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RepurposeResponse {
    pub content_input_id: String,
    pub outputs: Vec<RepurposedOutput>,
}
